cmd/gateway-gcs.go

```diff
@@ -835,7 +835,14 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 
 	partZero.Delete(l.ctx)
 
-	return fromGCSObjectInfo(attrs), gcsToObjectError(traceError(err), bucket, key)
+	objectInfo := fromGCSObjectInfo(attrs)
+	if objectInfo.ETag == "" {
+		h := md5.New()
+		io.WriteString(h, objectInfo.Bucket+objectInfo.Name+strconv.FormatInt(objectInfo.Size, 10))
```

Comment:
May be a simple strconv.Atoi would work? or perhaps just use `fmt.Sprintf("%s-%s-%d")`. 

One fundamental problem i see with this approach is that ETag needs to be persistent, if we cannot say that this value is persistent i.e say when 

GetObject() asks for `If-Match`  then we are in trouble. Can you check what do we respond for a GetObject() or a HeadObject() in case of GCS, we can basically use the same ETag i.e sent back to the client.